File: Taller de Haskell/Haskell/problem6.hs
asignarGenero :: String -> Char -> String
asignarGenero nombre genero
    | genero == 'M' || genero == 'm' = "Hombre: " ++ nombre
    | genero == 'F' || genero == 'f' = "Mujer: " ++ nombre
    | otherwise = "Genero no válido."

main :: IO ()
main = do
    putStrLn "Ingrese su nombre: "
    nombre <- getLine
    putStrLn "Ingrese su genero (M/F): "
    generoStr <- getLine
    let genero = if null generoStr then ' ' else head generoStr
    putStrLn (asignarGenero nombre genero)

File: Taller de Haskell/Python/problem6.py
# Asignación grupal

nombre = input("Ingrese su nombre: ")
genero = input("Ingrese su genero (M/F): ")

if genero == "M" or genero == "m":
    print("Hombre:", nombre)
elif genero == "F" or genero == "f":
    print("Mujer:", nombre)
else:
    print("Genero no válido.")